configure.ac

```diff
@@ -1042,11 +1043,11 @@ AC_SUBST([CXX])
 preset_cxx="$CXX"
 if test -z "$CXX"
 then
-        case "$CC" in
+        case "$CC_BASENAME" in
         gcc)    AC_PATH_TOOL([CXX], [g++], [g++], [notfound]) ;;
         cc)     AC_PATH_TOOL([CXX], [c++], [c++], [notfound]) ;;
-        clang|*/clang)     AC_PATH_TOOL([CXX], [clang++], [clang++], [notfound]) ;;
-        icc|*/icc)         AC_PATH_TOOL([CXX], [icpc], [icpc], [notfound]) ;;
+        clang)             AC_PATH_TOOL([CXX], [clang++], [clang++], [notfound]) ;;
+        icc)               AC_PATH_TOOL([CXX], [icpc], [icpc], [notfound]) ;;
```

Comment:
This doesn't seem to correspond to the definition of [AC_PATH_TOOL](https://www.gnu.org/software/autoconf/manual/autoconf-2.68/html_node/Generic-Programs.html):

`AC_PATH_TOOL (variable, prog-to-check-for, [value-if-not-found], [path = ‘$PATH’])`

* variable = CXX
* prog-to-check-for = g++
* value-if-not-found = g++ (?)
* path = notfound (??)

Reply:
Also, CC_BASENAME may contain a platform triplet, so shouldn't this use ac_cv_cc_name instead?

Reply:
> Also, CC_BASENAME may contain a platform triplet, so shouldn't this use ac_cv_cc_name instead?

Yes, this should use `ac_cv_cc_name`, not `CC_BASENAME`.

> This doesn't seem to correspond to the definition of [AC_PATH_TOOL](https://www.gnu.org/software/autoconf/manual/autoconf-2.68/html_node/Generic-Programs.html):

Indeed; I suggest we look at this in a separate PR.

Reply:
> Also, CC_BASENAME may contain a platform triplet, so shouldn't this use ac_cv_cc_name instead?

Addressed with 9900633bbfec69bf2526b01f13e1926cb0c9f750.